vaxrank/epitope_prediction.py

```diff
@@ -77,34 +87,101 @@ def logistic_score(
 
         return logistic / normalizer
 
-def predict_epitopes(mhc_predictor, protein_fragment, min_epitope_score=0):
-    results = []
+def fm_index_path(genome):
+    """
+    Returns a path for cached reference peptides, for the given genome.
+    """
+    cache_dir = user_cache_dir('vaxrank')
+    if not os.path.exists(cache_dir):
+        os.makedirs(cache_dir)
+
+    return os.path.join(cache_dir, '%s_%d.fm' % (
+        genome.species.latin_name, genome.release))
+
+def index_contains_kmer(fm, kmer):
+    """
+    Checks FM index for kmer, returns true if found.
+    """
+    found = False
+    for _ in fm.search(kmer):
+        found = True
+        break
+    return found
+
+def load_reference_peptides_index(genome, force_reload=False):
+    """
+    Loads the FM index containing reference peptides.
+
+    Parameters
+    ----------
+    genome : pyensembl.EnsemblRelease
+        Input genome to load for reference peptides
+
+    force_reload : bool, optional
+        If true, will recompute index for this genome even if it already exists.
+
+    Returns
+    -------
+    fm : shellinford.FMIndex
+        Index populated with reference peptides from the genome
+    """
+    path = fm_index_path(genome)
+    if force_reload or not os.path.exists(path):
+        logger.info("Loading FM index to %s", path)
+        fm = shellinford.FMIndex()
+        fm.build(
+            (t.protein_sequence for t in genome.transcripts() if t.is_protein_coding),
+            path)
+        logger.info("Done.")
```

Comment:
When you're logging "Done" any chance you could include what got done (e.g. "Done building FM index")? Sometimes I see those in the logs and it's not obvious to me.